src/core/namespace.h

```diff
@@ -72,9 +72,51 @@ struct TemporaryFileSystem {
         char *options;
 };
 
+typedef enum MountMode {
+        /* This is ordered by priority! */
+        INACCESSIBLE,
+        MOUNT_IMAGES,
+        BIND_MOUNT,
+        BIND_MOUNT_RECURSIVE,
+        PRIVATE_TMP,
+        PRIVATE_TMP_READONLY,
+        PRIVATE_DEV,
+        BIND_DEV,
+        EMPTY_DIR,
+        SYSFS,
+        PROCFS,
+        READONLY,
+        READWRITE,
+        TMPFS,
+        READWRITE_IMPLICIT, /* Should have the lowest priority. */
+        _MOUNT_MODE_MAX,
+} MountMode;
+
+typedef struct MountEntry {
+        const char *path_const;   /* Memory allocated on stack or static */
+        MountMode mode:5;
+        bool ignore:1;            /* Ignore if path does not exist? */
+        bool has_prefix:1;        /* Already is prefixed by the root dir? */
+        bool read_only:1;         /* Shall this mount point be read-only? */
+        bool nosuid:1;            /* Shall set MS_NOSUID on the mount itself */
+        bool applied:1;           /* Already applied */
+        char *path_malloc;        /* Use this instead of 'path_const' if we had to allocate memory */
+        const char *source_const; /* The source path, for bind mounts or path images */
+        char *source_malloc;
+        const char *options_const;/* Mount options for tmpfs */
+        char *options_malloc;
+        unsigned long flags;      /* Mount flags used by EMPTY_DIR and TMPFS. Do not include MS_RDONLY here, but please use read_only. */
+        unsigned n_followed;
+} MountEntry;
+
+const char *mount_entry_path(const MountEntry *p);
+const char *mount_entry_source(const MountEntry *p);
```

Comment:
hmm, I am not too fond of exporting this. This is so much internal details stuff. Would prefer to keep this internal

Reply:
As far as I can see, given there's a triplet source:dest:ignore the other option is to add a struct YetAnotherMountEntry, but to me it looked cleaner to just export and use the existing one and avoid the duplication. If you prefer the other approach, or another that I haven't thought about, I can do that.

Reply:
I'll see if I can use lists + helpers + opaque struct to avoid exposing the detailed struct

Reply:
@poettering so I've changed this to use list.h and accessors plus an opaque struct, so that the internal details are not exposed outside of namespace.h. Please let me know if this works better or if you'd like a different solution.
(I cherry-picked the LIST_JOIN commit from https://github.com/systemd/systemd/pull/16308 )

Reply:
Changed again to use its own struct MountImage.